generators/generator-base.js

```diff
@@ -1447,7 +1460,19 @@ module.exports = class extends PrivateBase {
      * @param {string} version - (optional) explicit dependency version number
      */
     addGradleDependency(scope, group, name, version) {
-        const fullPath = 'build.gradle';
+        this.addGradleDependencyInDirectory('.', scope, group, name, version);
+    }
+
+    /**
+     * A new dependency to build.gradle file in a specific folder.
+     *
+     * @param {string} scope - scope of the new dependency, e.g. compile
+     * @param {string} group - maven GroupId
+     * @param {string} name - maven ArtifactId
+     * @param {string} version - (optional) explicit dependency version number
+     */
+    addGradleDependencyInDirectory(directory, scope, group, name, version) {
+        const fullPath = `${directory}/build.gradle`;
```

Comment:
@deepu105 Maybe the use of the `path` module would be good here, don't you think?